tsdb/querier/querier.go

```diff
@@ -88,9 +88,10 @@ func (q *Query) Exec() []byte {
 
 // ExecWithoutEncode executes the Query without encoding the result to []byte,
 // rather keeps the result as default QueryResponse.
-func (q *Query) ExecWithoutEncode() QueryResponse {
+func (q *Query) ExecWithoutEncode(newBlocks []tsdb.Block) QueryResponse {
```

Comment:
If you need blocks to be passed for querying, we should have a `FillDataBlocks()` that should do this job. The param here is not defining anything IMO.

Reply:
Actually, earlier in `ExecWithoutEncode`, we reloaded all the blocks from the DBpath and ran `exec` function. Now I am passing down the newBlocks(in-memory chain), which is then appended to the fetched blocks from the local storage, then we are executing `exec`. Please explain how can I implement this otherwise.